Source/Scene/ModelExperimental/Cesium3DTileContentFeatureTable.js

```diff
@@ -0,0 +1,170 @@
+ * The properties for a feature are extracted from a {@link FeatureTable}.
+ *
+ * @param {Object} options An object containing the following options:
+ * @param {Cesium3DTileContent} options.content The tile content this features in this table belong to.
+ * @param {FeatureTable} options.featureTable The feature table from the model belonging to the content.
+ *
+ * @alias Cesium3DTileContentFeatureTable
+ * @constructor
+ * @private
+ * @experimental This feature is using part of the 3D Tiles spec that is not final and is subject to change without Cesium's standard deprecation policy.
+ */
+export default function Cesium3DTileContentFeatureTable(options) {
+  var content = options.content;
+  var featureTable = options.featureTable;
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.content", content);
+  Check.typeOf.object("options.featureTable", featureTable);
+  //>>includeEnd('debug');
+
+  this._content = content;
+  this._featureTable = featureTable;
+
+  this._featuresLength = 0;
+  this._features = undefined;
+
+  this._batchTexture = undefined;
+
+  initialize(this);
+}
+
+Object.defineProperties(Cesium3DTileContentFeatureTable.prototype, {
+  /**
+   * The batch texture generated for the features in this table.
+   *
+   * @memberof Cesium3DTileContentFeatureTable.prototype
+   * @type {BatchTexture}
+   * @readonly
+   * @private
+   */
+  batchTexture: {
+    get: function () {
+      return this._batchTexture;
+    },
+  },
+  /**
+   * The number of features in this table.
+   *
+   * @memberof Cesium3DTileContentFeatureTable.prototype
+   * @type {Number}
+   * @readonly
+   * @private
+   */
+  featuresLength: {
+    get: function () {
+      return this._featuresLength;
+    },
+  },
+});
+
+function initialize(contentFeatureTable) {
+  var featuresLength = contentFeatureTable._featureTable.count;
+  if (featuresLength === 0) {
+    return;
+  }
+
+  var content = contentFeatureTable._content;
+
+  var features = new Array(featuresLength);
+  for (var i = 0; i < featuresLength; i++) {
+    features[i] = new Cesium3DTileFeature(content, i);
+  }
+
+  contentFeatureTable._featuresLength = featuresLength;
+  contentFeatureTable._features = features;
+
+  contentFeatureTable._batchTexture = new BatchTexture({
+    featuresLength: featuresLength,
+    owner: content,
+    statistics: content.tileset.statistics,
+  });
+}
+
+Cesium3DTileContentFeatureTable.prototype.getFeature = function (featureId) {
+  return this._features[featureId];
+};
+
+Cesium3DTileContentFeatureTable.prototype.hasProperty = function (
+  featureId,
+  propertyName
+) {
+  return this._featureTable.hasProperty(featureId, propertyName);
+};
+
+Cesium3DTileContentFeatureTable.prototype.getProperty = function (
+  featureId,
+  name
+) {
+  return this._featureTable.getProperty(featureId, name);
+};
+
+Cesium3DTileContentFeatureTable.prototype.getPropertyInherited = function (
```

Comment:
In this new architecture, is it still `Cesium3DTileFeature` that handles inheriting tileset/group/tile metadata? if so, do the feature tables need this method?

Reply:
I think we'll handle this at the `Cesium3DTileContent` or in the `Cesium3DTileContentFeatureTable` level once we get to B3DM. Probably the former.